mempool/src/config.rs

```diff
@@ -46,6 +47,61 @@ impl std::fmt::Display for MempoolMaxSize {
     }
 }
 
+impl serde::Serialize for MempoolMaxSize {
+    fn serialize<S: serde::Serializer>(&self, serializer: S) -> Result<S::Ok, S::Error> {
+        serializer.serialize_u64(self.0 as u64)
+    }
+}
+
+#[derive(serde::Serialize, serde::Deserialize)]
+#[serde(untagged)]
+enum StringOrUint {
+    String(String),
+    UInt(u64),
+}
+
+impl<'de> serde::Deserialize<'de> for MempoolMaxSize {
+    fn deserialize<D: serde::Deserializer<'de>>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        use serde::de::Error;
+
+        match StringOrUint::deserialize(deserializer)? {
+            StringOrUint::String(s) => {
+                let bytes: byte_unit::Byte = rpc::handle_result(byte_unit::Byte::from_str(&s))
+                    .map_err(|e| {
+                        D::Error::custom(format!(
+                            "Conversion to integer from data size failed: {e}"
+                        ))
+                    })?;
+
+                let bytes: usize = bytes.as_u64().try_into().map_err(|_| {
+                    D::Error::custom(format!("Integer {} is too large to fit", bytes))
+                })?;
+
+                let max_size = MempoolMaxSize::from_bytes(bytes);
+
+                Ok(max_size)
+            }
+            StringOrUint::UInt(bytes) => {
+                let bytes: usize = bytes.try_into().map_err(|_| {
+                    D::Error::custom(format!("Integer {} is too large to fit", bytes))
+                })?;
+
+                let max_size = MempoolMaxSize::from_bytes(bytes);
+
+                Ok(max_size)
+            }
+        }
+    }
+}
+
+impl HasValueHint for MempoolMaxSize {
+    const HINT: rpc_description::ValueHint =
+        rpc_description::ValueHint::Prim("Size with units, such as MB/KB/GB, or integer for bytes");
```

Comment:
This is probably too long to be a value hint. It is included in the params in the generated docs and is expected to be rather short as it can be a part of a more complicated type description. Have you checked how it looks in the generated doc? It may be better just have some sort of succinct name here and add details to a doc comment somewhere.

Reply:
I could use that to simplify the implementation. I'll give it a shot. But this doesn't provide the features we need. We need the serialization to always lead to a number, while this doesn't. Also we want this to be without a key, so simple deriving doesn't work.

Reply:
I seem to have commented on the wrong comment. This is about serde.

Reply:
Just checked. This looks fine.